pkg/cmd/server/origin/master.go

```diff
@@ -314,7 +319,13 @@ func (c *MasterConfig) buildHandlerChain(assetConfig *AssetConfig) (func(http.Ha
 
 		handler = kgenericfilters.WithCORS(handler, c.Options.CORSAllowedOrigins, nil, nil, nil, "true")
 		handler = kgenericfilters.WithPanicRecovery(handler, c.RequestContextMapper)
-		handler = kgenericfilters.WithTimeoutForNonLongRunningRequests(handler, kc.LongRunningFunc)
+
+		longRunningRequestRE := regexp.MustCompile(longRunningEndpointsRE)
+		longRunningRequestCheck := kgenericfilters.BasicLongRunningRequestCheck(longRunningRequestRE, nil)
+		longRunningRequestCheckFunc := func(r *http.Request) bool {
+			return longRunningRequestCheck(r) || kc.LongRunningFunc(r)
+		}
+		handler = kgenericfilters.WithTimeoutForNonLongRunningRequests(handler, longRunningRequestCheckFunc)
```

Comment:
Can you elaborate on why you don't like it here?  This puts it as close to where we setup the handler as possible, ensuring it doesn't get skipped by an alternate codepath (e.g. test code that mocks bringing up the api server).